build.sh

```diff
@@ -35,67 +72,113 @@ do
 fi
 
 # XXX should also check if .entity-processor.py, .entity-processor-json.py, and entities-legacy* have changed
-if [ unicode.xml -nt entities-unicode.inc ]; then
+if [ $HTML_CACHE/unicode.xml -nt $HTML_TEMP/entities-unicode.inc ]; then
   $QUIET || echo;
-  $QUIET || echo "Updating entities database...";
-  python .entity-processor.py > .new-entities-unicode.inc;
-  [ -s .new-entities-unicode.inc ] && mv -f .new-entities-unicode.inc entities-unicode.inc; # otherwise, probably http error, just do it again next time
-  python .entity-processor-json.py > .new-entities-unicode-json.inc;
-  [ -s .new-entities-unicode-json.inc ] && mv -f .new-entities-unicode-json.inc json-entities-unicode.inc; # otherwise, probably http error, just do it again next time
-  echo '<tbody>' > entities.inc
-  rm -f entities-dtd.url entities*~ json-*~ json-entities.inc
-  cat entities-*.inc | perl -e 'my @lines = <>; print sort { $a =~ m/id="([^"]+?)(-legacy)?"/; $a1 = $1; $a2 = $2; $b =~ m/id="([^"]+?)(-legacy)?"/; $b1 = $1; $b2 = $2; return (lc($a1) cmp lc($b1)) || ($a1 cmp $b1) || ($a2 cmp $b2); } @lines' >> entities.inc
-  echo '{' > entities.json
-  cat json-entities-* | sort | perl -e '$/ = undef; $_ = <>; chop, chop, print' >> entities.json
-  echo '' >> entities.json
-  echo '}' >> entities.json
-  perl -Tw .entity-to-dtd.pl < entities-unicode.inc > entities-dtd.url
+  $QUIET || echo "Updating entities database (this always takes a while)...";
+  python .entity-processor.py > $HTML_TEMP/new-entities-unicode.inc;
+  [ -s $HTML_TEMP/new-entities-unicode.inc ] && mv -f $HTML_TEMP/new-entities-unicode.inc $HTML_TEMP/entities-unicode.inc; # otherwise, probably http error, just do it again next time
+  python .entity-processor-json.py > $HTML_TEMP/new-entities-unicode-json.inc;
+  [ -s $HTML_TEMP/new-entities-unicode-json.inc ] && mv -f $HTML_TEMP/new-entities-unicode-json.inc $HTML_TEMP/json-entities-unicode.inc; # otherwise, probably http error, just do it again next time
+  echo '<tbody>' > $HTML_TEMP/entities.inc
+  rm -f $HTML_TEMP/entities-dtd.url $HTML_TEMP/entities*~ $HTML_TEMP/json-*~ $HTML_TEMP/json-entities.inc
+  cat $HTML_TEMP/entities-*.inc | perl -e 'my @lines = <>; print sort { $a =~ m/id="([^"]+?)(-legacy)?"/; $a1 = $1; $a2 = $2; $b =~ m/id="([^"]+?)(-legacy)?"/; $b1 = $1; $b2 = $2; return (lc($a1) cmp lc($b1)) || ($a1 cmp $b1) || ($a2 cmp $b2); } @lines' >> $HTML_TEMP/entities.inc
+  echo '{' > $HTML_OUTPUT/entities.json
+  cat $HTML_TEMP/json-entities-* | sort | perl -e '$/ = undef; $_ = <>; chop, chop, print' >> $HTML_OUTPUT/entities.json
+  echo '' >> $HTML_OUTPUT/entities.json
+  echo '}' >> $HTML_OUTPUT/entities.json
+  perl -Tw .entity-to-dtd.pl < $HTML_TEMP/entities-unicode.inc > $HTML_TEMP/entities-dtd.url
 fi
 
-if [ "$DO_UPDATE" == true ] || [ ! -f caniuse.json ] || [ ! -f w3cbugs.csv ]; then
-  rm -f caniuse.json w3cbugs.csv
+if [ "$DO_UPDATE" == true ] || [ ! -f $HTML_CACHE/caniuse.json ] || [ ! -f $HTML_CACHE/w3cbugs.csv ]; then
```

Comment:
> Maybe separate this out into two ifs, so that someone could delete caniuse.json from their cache but not re-download the bug database?

OK, yup—done.